clients/go/compat.go

```diff
@@ -31,283 +35,241 @@ type NetworkParams struct {
+	return b.backend.SuggestGasPrice(ctx)
 }
 
-func (b WrappedBackend) CodeAt(ctx context.Context, contract common.Address, blockNumber *big.Int) ([]byte, error) {
-	return b.Backend.CodeAt(ctx, contract, blockNumber)
+// SuggestGasTipCap implemennts ContractTransactor.
+func (b WrappedBackend) SuggestGasTipCap(ctx context.Context) (*big.Int, error) {
+	return b.backend.SuggestGasTipCap(ctx)
 }
 
-// Return Sapphire default gas limit.
+// EstimateGas implements ContractTransactor.
 func (b WrappedBackend) EstimateGas(ctx context.Context, call ethereum.CallMsg) (gas uint64, err error) {
 	return DefaultGasLimit, nil
-	// TODO: re-enable
-	// header, err := b.Backend.HeaderByNumber(ctx, blockNumber)
-
+	// TODO(#39)
+	// header, err := b.backend.HeaderByNumber(ctx, blockNumber)
 	// if err != nil {
 	// 	return nil, err
 	// }
-
 	// if err = packContractCall(header, &call, b.ChainID.Uint64(), b.Signer, b.Cipher); err != nil {
 	// 	return nil, err
 	// }
-
-	// return b.Backend.EstimateGas(ctx, call)
-}
-
-func (b WrappedBackend) FilterLogs(ctx context.Context, query ethereum.FilterQuery) ([]types.Log, error) {
-	return b.Backend.FilterLogs(ctx, query)
-}
-
-func (b WrappedBackend) HeaderByNumber(ctx context.Context, number *big.Int) (*types.Header, error) {
-	return b.Backend.HeaderByNumber(ctx, number)
+	// return b.backend.EstimateGas(ctx, call)
 }
 
-func (b WrappedBackend) PendingCodeAt(ctx context.Context, account common.Address) ([]byte, error) {
-	return b.Backend.PendingCodeAt(ctx, account)
-}
-
-func (b WrappedBackend) PendingNonceAt(ctx context.Context, account common.Address) (uint64, error) {
-	return b.Backend.PendingNonceAt(ctx, account)
+func txNeedsPacking(tx *types.Transaction) bool {
+	_, isPacked := packedTxs.Load(tx.Hash())
+	if tx == nil || len(tx.Data()) == 0 || isPacked {
+		return false
+	}
+	var envelope Data
+	return cbor.Unmarshal(tx.Data(), &envelope) != nil // If there is no error, the tx is already packed.
 }
 
-// Return Sapphire default gas price.
-func (b WrappedBackend) SuggestGasPrice(ctx context.Context) (*big.Int, error) {
-	return big.NewInt(DefaultGasPrice), nil
-	// TODO: re-enable
-	// return b.Backend.SuggestGasPrice(ctx)
+// SendTransaction implements ContractTransactor.
+func (b WrappedBackend) SendTransaction(ctx context.Context, tx *types.Transaction) error {
+	if !txNeedsPacking(tx) {
+		return b.backend.SendTransaction(ctx, tx)
+	}
+	packedTx, err := packTx(*tx, b.cipher)
+	if err != nil {
+		return fmt.Errorf("failed to pack tx: %w", err)
+	}
+	signer := types.LatestSignerForChainID(&b.chainID)
+	txHash := signer.Hash(packedTx).Bytes()
+	signature, err := b.sign(*(*[32]byte)(txHash))
+	if err != nil {
+		return fmt.Errorf("failed to sign wrapped tx: %w", err)
+	}
+	signedTx, err := packedTx.WithSignature(signer, signature)
+	if err != nil {
+		return fmt.Errorf("failed to attach signature to wrapped tx: %w", err)
+	}
+	packedTxs.Delete(signedTx.Hash())
```

Comment:
Why do we have to delete this hash?

Reply:
to prevent memory use from growing without bound